File: contest/ABC/164/question-d.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(){
    char s[200000], t[200000];
    int i, j, k, d, cnt;
    scanf("%s", &s);
    cnt = 0;
    for( i=0; i<strlen(s)-3; i++ ){
        for( j=i+3; j<strlen(s); j++ ){
            d = 0;
            for( k=0; k<j-i; k++ ){
                d += pow(10,k) * s[k];
                printf("%d\n", d);
            }
            if( d%2019==0 ){
                cnt++;
            }     
        }
    }
    printf("%d\n", cnt);
    return 0;
}

File: contest/ABC/164/question-d.py
s = input()

cnt = 0
#for c in range( 50 ): #int(s) // 2019 ):
#    d = ( c + 1 ) * 2019
#    for n in range( len( str(s) ) - len( str(d) ) ):
#        if str( d ) in str( s )[n:]:
#            cnt += 1
#            print( d )
#            break

for i in range( len(s) - 3 ):
    for j in range( i+3, len(s) ):
        d = int( s[i:j+1] )
        if d % 2019 == 0:
            cnt += 1
            #print(i,j)

print( cnt )